internal/resolution/datasource/maven_registry.go

```diff
@@ -27,50 +33,101 @@ type MavenRegistryAPIClient struct {
 	metadata       *RequestCache[string, maven.Metadata]
 }
 
-func NewMavenRegistryAPIClient(registry string) *MavenRegistryAPIClient {
+func NewMavenRegistryAPIClient(registry string) (*MavenRegistryAPIClient, error) {
+	if registry == "" {
+		registry = MavenCentral
+	} else if _, err := url.Parse(registry); err != nil {
+		return nil, fmt.Errorf("invalid Maven registry %s: %w", registry, err)
+	}
+
 	return &MavenRegistryAPIClient{
-		registry: registry,
-		projects: NewRequestCache[string, maven.Project](),
-		metadata: NewRequestCache[string, maven.Metadata](),
+		defaultRegistry: registry,
+		projects:        NewRequestCache[string, maven.Project](),
+		metadata:        NewRequestCache[string, maven.Metadata](),
+	}, nil
+}
+
+// Add adds the given registry to the list of registries if it has not been added.
+func (m *MavenRegistryAPIClient) AddRegistry(registry string) error {
+	if !slices.Contains(m.registries, registry) {
```

Comment:
nit: it seems better to invert this condition and return nil instead, to avoid more indentation on the common case.